src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3345,6 +3330,15 @@ private static Symbol AsMemberOfType(NamedTypeSymbol containingType, Symbol symb
                     {
                         return AsMemberOfTupleType((TupleTypeSymbol)containingType, symbol);
                     }
+                    if (symbolDefContainer.IsAnonymousType)
+                    {
+                        if (symbol.Kind != SymbolKind.Property)
+                        {
+                            Debug.Assert(false);
```

Comment:
Nit: could use `Debug.Fail` with an error message.

Reply:
Alternatively remove the `if` and use `Debug.Assert(symbol.Kind == SymbolKind.Property)`. Would it ever be reasonable to just `break` here in a release build but fail in a debug build?

Reply:
Added a comment rather than an error message. 

---
In reply to: [259068854](https://github.com/dotnet/roslyn/pull/33562#discussion_r259068854) [](ancestors = 259068854)

Reply:
We should handle this case if the assert fails. For now, if we hit this code path in a release build, we'll return the original symbol which should allow continuing, although perhaps with missing warnings.

---
In reply to: [259073346](https://github.com/dotnet/roslyn/pull/33562#discussion_r259073346) [](ancestors = 259073346)